src/Features/Lsif/Generator/Generator.cs

```diff
@@ -141,9 +143,11 @@ public async Task GenerateForCompilationAsync(Compilation compilation, string pr
                 // We always use UTF-8 encoding when writing out file contents, as that's expected by LSIF implementations.
                 // TODO: when we move to .NET Core, is there a way to reduce allocations here?
                 contentBase64Encoded = Convert.ToBase64String(Encoding.UTF8.GetBytes(text.ToString()));
+
+                uri = "source-generated://" + syntaxTree.FilePath.Replace('\\', '/');
```

Comment:
I think it would be good to land on a consistent URI scheme for source generated documents

For example currently LSP does this - https://github.com/dotnet/roslyn/blob/main/src/Features/LanguageServer/Protocol/Extensions/ProtocolConversions.cs#L33

Reply:
@dibarbet, ah right my commit message of "we should figure a standard" didn't get updated here. :smile:

The bigger thing we need is not just the prefix, but also what the other parts of the URI are as well, since we also need something to differentiate different projects/targets. Did you do anything for that?

Reply:
Not yet - the general idea would be to encode them as parameters.

Perhaps its OK to wait to unify until we start actually doing the virtual docs support. But maybe at least for now we should use the same prefix

Reply:
I chose source-generated rather than just "generated" to keep the scheme more explicit, since we may also have other schemes like metadata-as-source or source-link or something else and those need to go to different LSP handlers or something.